### PR TITLE
[Feature][Shade] Add seatunnel hadoop3 uber

### PR DESCRIPTION
**Pull request by EricJoy2048**
Now many module test used `flink shade hadoop`, We tested use `flink shade hadoop` jar will trigger jackson jar conflict when we use `MySQL CDC Source`. So we add a seatunnel owner hadoop shaded uber jar.

### CODE REVIEW

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-iceberg-flink-e2e/pom.xml [hunk 1]
```diff
@@ -41,8 +41,8 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.flink</groupId>
-            <artifactId>flink-shaded-hadoop-2</artifactId>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-hadoop3-uber</artifactId>
             <scope>test</scope>
```
Comment (Grypse) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-iceberg-flink-e2e/pom.xml:48]: Need version scope.

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/pom.xml [hunk 2]
```diff
@@ -52,8 +52,8 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.flink</groupId>
-            <artifactId>flink-shaded-hadoop-2</artifactId>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-hadoop3-uber</artifactId>
             <scope>test</scope>
```
Comment (Grypse) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/pom.xml:59]: Need version scope.

File: seatunnel-engine/seatunnel-engine-client/pom.xml [hunk 3]
```diff
@@ -79,8 +79,8 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.flink</groupId>
-            <artifactId>flink-shaded-hadoop-2</artifactId>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-hadoop3-uber</artifactId>
             <scope>test</scope>
```
Comment (Grypse) [inline seatunnel-engine/seatunnel-engine-client/pom.xml:86]: Need version scope.

File: seatunnel-engine/seatunnel-engine-server/pom.xml [hunk 4]
```diff
@@ -82,8 +82,8 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.flink</groupId>
-            <artifactId>flink-shaded-hadoop-2</artifactId>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-hadoop3-uber</artifactId>
             <scope>test</scope>
```
Comment (Grypse) [inline seatunnel-engine/seatunnel-engine-server/pom.xml:94]: Need version scope.

**Review by Grypse — CHANGES_REQUESTED** [review]
Add version scope to seatunnel uber dependency.

File: seatunnel-shade/seatunnel-hadoop3-uber/pom.xml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>seatunnel-hadoop3-uber</artifactId>
+
+    <dependencies>
+        <!-- https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-aws -->
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-client</artifactId>
+            <version>3.1.4</version>
```
Comment (TyrantLucifer) [inline seatunnel-shade/seatunnel-hadoop3-uber/pom.xml]:
```suggestion
            <version>${hadoop.version}</version>
```

File: seatunnel-engine/seatunnel-engine-server/pom.xml (hunk 4 above)
Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-server/pom.xml:94]: done

File: seatunnel-engine/seatunnel-engine-client/pom.xml (hunk 3 above)
Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-client/pom.xml:86]: done

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/pom.xml (hunk 2 above)
Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/pom.xml:59]: done

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-iceberg-flink-e2e/pom.xml (hunk 1 above)
Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-iceberg-flink-e2e/pom.xml:48]: done

**Comment by Grypse** [comment]
Please check CI  errors.

File: seatunnel-shade/pom.xml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        <artifactId>seatunnel</artifactId>
+        <groupId>org.apache.seatunnel</groupId>
+        <version>${revision}</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>seatunnel-shade</artifactId>
+    <packaging>pom</packaging>
+    <modules>
+        <module>seatunnel-hadoop3-3.1.4-uber</module>
```
Comment (Grypse) [inline seatunnel-shade/pom.xml:29]: I think the name of seatunnel-hadoop3-shade may be better.There's no need to add hadoop version to  module name.

File: seatunnel-shade/seatunnel-hadoop3-3.1.4-uber/pom.xml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>seatunnel-hadoop3-3.1.4-uber</artifactId>
+
+    <dependencies>
+        <!-- https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-aws -->
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-client</artifactId>
+            <version>3.1.4</version>
```
Comment (Grypse) [inline seatunnel-shade/seatunnel-hadoop3-3.1.4-uber/pom.xml]: To manage the version with properties may be better.

Comment (EricJoy2048) [reply seatunnel-shade/seatunnel-hadoop3-3.1.4-uber/pom.xml]: done

File: seatunnel-shade/pom.xml (hunk 6 above)
Comment (EricJoy2048) [reply seatunnel-shade/pom.xml:29]: > I think the name of seatunnel-hadoop3-shade may be better.There's no need to add hadoop version to module name.

I don't think so. We will add more hadoop version shaded jar in the next version.

File: seatunnel-shade/seatunnel-hadoop3-uber/pom.xml (hunk 5 above)
Comment (EricJoy2048) [reply seatunnel-shade/seatunnel-hadoop3-uber/pom.xml]: done

**Review by TyrantLucifer — APPROVED** [review]

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-iceberg-flink-e2e/pom.xml [hunk 8]
```diff
@@ -41,8 +41,10 @@
...
         </dependency>
         <dependency>
-            <groupId>org.apache.flink</groupId>
-            <artifactId>flink-shaded-hadoop-2</artifactId>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-hadoop3-3.1.4-uber</artifactId>
+            <version>${project.version}</version>
+            <classifier>optional</classifier>
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-iceberg-flink-e2e/pom.xml:47]: Mybe `<optional>true</optional>`?

File: seatunnel-engine/seatunnel-engine-client/pom.xml [hunk 9]
```diff
@@ -79,8 +79,10 @@
...
         </dependency>
         <dependency>
-            <groupId>org.apache.flink</groupId>
-            <artifactId>flink-shaded-hadoop-2</artifactId>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-hadoop3-3.1.4-uber</artifactId>
+            <version>${project.version}</version>
+            <classifier>optional</classifier>
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-client/pom.xml:85]: Mybe `<optional>true</optional>` ?

File: seatunnel-engine/seatunnel-engine-server/pom.xml [hunk 10]
```diff
@@ -86,7 +86,13 @@
             <version>${project.version}</version>
             <scope>test</scope>
         </dependency>
-
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-hadoop3-3.1.4-uber</artifactId>
+            <version>${project.version}</version>
+            <classifier>optional</classifier>
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-server/pom.xml:93]: Mybe `<optional>true</optional>` ?

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml [hunk 11]
```diff
@@ -31,16 +31,11 @@
...
-        </dependency>
-        <dependency>
-            <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-client</artifactId>
-            <version>3.0.0</version>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-hadoop3-3.1.4-uber</artifactId>
+            <version>${project.version}</version>
             <scope>provided</scope>
+            <classifier>optional</classifier>
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml:38]: Mybe `<optional>true</optional>` ?

Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml:38]: > <optional>true</optional>

I don't understand what you mean. `optional` is a classifier, not a dependency parameter item.

**Review by liugddx — APPROVED** [review]
LGTM

**Pull request merged**